proposals/20210707-xgboost_evaluator.md

```diff
@@ -0,0 +1,136 @@
+
+Our custom prediction extractor essentially governs conversion of data to formats that xgboost can accept, extraction of the necessary features of the data, the actual prediction, and framework code supporting all of these operations. It will be passed into (this)[https://tensorflow.google.cn/tfx/model_analysis/api_docs/python/tfma/default_extractors] `tfma.default_extractors` function for use in the Evaluator.
+
+Currently, we plan to support running the Evaluator with Apache Beam through the use of a customized prediction DoFn to load, process, and run predictions on models, and a simple pipeline wrapper that calls this function on extracts.
+
+The actual implementation of the custom prediction extractor depends on whether it receives a native XGBoost serialized model (option 1 from above), or a pickled sklearn Pipeline (option 2 from above). Here are pros and cons of each option.
+
+**Option 1 (chosen)**:
+
+* Pros:
+  * Universal among the various XGBoost interfaces (Python, JVM, C++, etc.)
+  * Some level of backward compatibility is guaranteed
+  * Still retain attributes such as feature_names, feature_types, etc. (in newer xgboost versions)
+* Cons:
+  * Lack ability to combine with or substitute in other sklearn models
+
+Option 2:
+
+* Pros:
+  * Another wrapping layer means more flexibility, you can add some pre-processing and post-processing to the sklearn Pipeline, try out other types of models, etc.
+  * Most of the code needed for sklearn-compatible Trainer and Evaluator in the penguin sklearn pipeline can be reused
+
+* Cons:
+  * Extra dependency on sklearn
+  * Using Python pickle standard library or joblib, which is specific to Python
+  * Lack of guarantee for backward compatibility
+
+**Summary:** we will go with option 1 for simplicity and broader compatibility across different XGBoost interfaces.
+
+Open questions:
+From training performance view point, is there a difference between using native xgboost vs using sklearn Pipeline?
+
+### Testing and Example Pipeline
+
+In the same spirit as https://github.com/tensorflow/tfx-addons/blob/main/proposals/20210404-sklearn_example.md, we will add an example pipeline that run locally, and another version that runs on GCP using Vertex AI Pipelines.
+
+Example pipeline will have its end to end local unit test.
+
+The model can be pushed to CAIP. The current CAIP runtime version 2.5 runs XGBoost 1.4.0. Training, serialization, and deserialization XGBoost models using different versions of the library is allowed. In other words, the XGBoost library guarantees some level of backward compatibility.
+
+This example pipeline will not be packaged, instead, users just need to clone the source code to run the example.
+
+### Project directories
+
+* `tfx_addons/xgboost_evaluator` (evaluator code and tests)
+* `examples/xgboost_penguins`  (example pipelines and tests)
```

Comment:
Please reference the example from the component README so that folks are aware of it, and vice versa.